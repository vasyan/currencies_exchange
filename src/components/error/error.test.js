import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Error from './index'

describe('<Error>', () => {
  it('should render', () => {
    const props = {
      onClick: jest.fn()
    }

    expect(shallowToJson(shallow(<Error {...props} />))).toMatchSnapshot()
  })

  it('should call passed handler on click to "try again"', () => {
    const props = {
      onClick: jest.fn()
    }
    const wrapper = shallow(<Error {...props} />)

    wrapper.find('.try_again').simulate('click')

    expect(props.onClick).toHaveBeenCalled()
  })
})
