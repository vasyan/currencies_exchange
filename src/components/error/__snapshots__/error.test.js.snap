// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error> should render 1`] = `
<div
  className="wrapper"
>
  <div>
    <p
      className="alert"
    >
      Sorry, something 
      <br />
      has gone wrong.
    </p>
    <span
      className="try_again"
      onClick={[MockFunction]}
    >
      Try again
    </span>
  </div>
</div>
`;
